### DISCUSSION TITLE
useContract read is successful while useContractRead throws error

### DISCUSSION THREAD

**Discussion by matthewhausman** [wevm/wagmi#1484] [Q&A]
This is only an issue when not connected to wallet.

The following successfully reads the 'goal':

```
const saleContract = useContract({
    address: saleContractAddress,
    abi: saleAbi,
    signerOrProvider: provider
})

useEffect(() => {
    const asyncFn = async () => {
      const res = await saleContract?.goal()
      console.log(res) // Some Big Number
    }
    asyncFn()
}, [saleContract])
```

While the useContractRead hook will throw an error and look like: 

```
const { data: goalData } = useContractRead({
    enabled: !!saleContractAddress,
    address: saleContractAddress,
    abi: saleAbi,
    functionName: 'goal',
})
```

(Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="goal()", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.7.0))

**Comment by matthewhausman** [comment]
Solved :/

**Reply by cilphex** [reply]
What was the solution?

**Reply by matthewhausman** [reply]
add the chainId in useContractRead, if you're not connected the function isn't aware of target chain
